soroban-env-host/src/auth.rs

```diff
@@ -1760,14 +1750,13 @@ impl AccountAuthorizationTracker {
     ) -> Result<(), HostError> {
         self.invocation_tracker
             .rollback(&snapshot.invocation_tracker_root_snapshot)?;
-        self.authenticated = snapshot.authenticated;
-        self.need_nonce = snapshot.need_nonce;
+        self.verified = snapshot.verified;
         Ok(())
     }
 
     // metering: free
     fn is_active(&self) -> bool {
-        self.is_valid && self.invocation_tracker.is_active()
+        self.verified && self.invocation_tracker.is_active()
```

Comment:
Is this correct? I mean .. if it hasn't been verified _yet_, isn't it still potentially active?

Reply:
These conditions are mutually exclusive (because we only match invocations to valid trackers), so I guess `verified` check is kind of redundant.